cli/internal/core/engine.go

```diff
@@ -422,3 +431,45 @@ func (e *Engine) GetTaskGraphDescendants(taskID string) ([]string, error) {
 	sort.Strings(stringDescendents)
 	return stringDescendents, nil
 }
+
+func (e *Engine) getPipelineFromWorkspace(workspaceName string) (*fs.Pipeline, error) {
+	cachedPipeline, ok := e.pipelines[workspaceName]
+	if ok {
+		return cachedPipeline, nil
+	}
+
+	// Note: dir for the root workspace will be an empty string, and for
+	// other workspaces, it will be a relative path.
+	dir := e.completeGraph.WorkspaceInfos[workspaceName].Dir
+
+	var pkgJSON *fs.PackageJSON
+	repoRoot := e.completeGraph.RepoRoot
+
+	dirAbsolutePath := dir.RestoreAnchor(repoRoot)
+
+	// We need to get a package.json, because turbo.json can "synthesize" tasks from it
+	// for single-package repos.
+	if workspaceName == util.RootPkgName {
+		var err error
```

Comment:
Not a huge deal, but usually having to declare `err` like this is an indication that it would be better to use a different temp variable elsewhere. In this case it would be something instead of `pkgJSON`. The reason is that accidentally shadowing or reusing `err` is a somewhat common source of Go logic errors, so you almost never want to deviate from the standard pattern of `myVar, err := ...`